nat-lab/tests/conftest.py

```diff
@@ -67,6 +68,52 @@ def event_loop():
             loop.close()
 
 
+async def win_ports(vm_tag):
+    async def on_output(output: str) -> None:
+        print(f"win_ports_{vm_tag}: {output}")
+
+    start_port = random.randint(5000, 55000)
+    num_ports = random.randint(2000, 5000)
+    print(f"Setting up ports for {vm_tag}: start={start_port}, num={num_ports}")
+
+    async with new_connection_raw(vm_tag) as connection:
+        try:
+            await connection.create_process([
+                "netsh",
+                "int",
+                "ipv4",
+                "set",
+                "dynamic",
+                "tcp",
+                "start=" + str(start_port),
+                "num=" + str(num_ports),
+            ]).execute(on_output, on_output)
+        except ProcessExecError as e:
+            print(f"win_ports_{vm_tag} process execution failed: {e}")
+            return False
```

Comment:
nit: I'd suggest of getting rid return booleans and exception handling. Would move all exception handing in the fixture itself. That way it would be more python like.

Reply:
Fixed